### PR TITLE
Update ruff and clang-format versions

### PR DESCRIPTION
**Pull request by justinchuby**
ruff -> 0.5.4
clang-format -> 18

### CODE REVIEW

**Review by snnn — APPROVED** [review]

File: cgmanifests/generate_cgmanifest.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def add_github_dep(name, parsed_url):
             return
         # Make a REST call to convert to tag to a git commit
         url = f"https://api.github.com/repos/{org_name}/{repo_name}/git/refs/tags/{tag}"
-        print("requesting %s ..." % url)
+        print("requesting {url} ...")
```
Comment (snnn) [inline cgmanifests/generate_cgmanifest.py:76]: This change is not right. Now it prints the string as it is.
```
requesting {url} ...
```
@justinchuby

Comment (justinchuby) [reply cgmanifests/generate_cgmanifest.py:76]: I forgot an f""

Comment (justinchuby) [reply cgmanifests/generate_cgmanifest.py:76]: I will fix forward

Comment (justinchuby) [reply cgmanifests/generate_cgmanifest.py:76]: https://github.com/microsoft/onnxruntime/pull/21489

**Pull request merged**